basetest.pm

```diff
@@ -374,6 +376,15 @@ sub runtest {
         $died = 1;
     }
 
+    # Detect autoinst failures and override result if die
+    eval { $self->search_for_expected_autoinst_failures(); };
```

Comment:
I'd prefer use of `Try::Tiny` here over eval. (Using a return code would be sufficient here as well.)